### PR TITLE
Improve gui and raisd setup files and readme

### LINKED ISSUE #135: Improve automated gui setup
Change way setup is structured:

channel-priority: flexible is not accepted by conda, change the giving package manager to just default conda or if flag is given micromamba
add how to run gui to readme:
activate raisd-ai-gui
python -m gui.app
add alternative to commands in form of sh script to run gui.
activate environment before running compile raisd
change base raisd url & setup
add missing packages to raisd-ai environment

### PR DESCRIPTION
**Pull request by Steef-Broeder**
Closes #135.
both conda and micromamba have been tested and work.

### CODE REVIEW

**Comment by Steef-Broeder** [comment]
partial fix for png.h not being recognized + initializing conda
```bash
#!/bin/bash

INSTALL_ALL=false
PACKAGE_MANAGER="conda" # Default package manager
RAiSD_AI_VERSION="RAiSD-AI-ZLIB" # Default version

export C_INCLUDE_PATH=$CONDA_PREFIX/include:$C_INCLUDE_PATH
export LIBRARY_PATH=$CONDA_PREFIX/lib:$LIBRARY_PATH
export LD_LIBRARY_PATH=$CONDA_PREFIX/lib:$LD_LIBRARY_PATH

create_environment() {
    local env_file=$1

    echo "Creating environment from: $env_file"
    #$PACKAGE_MANAGER env create -f "$env_file"
}

install_raisd_ai() {
    # check for existance of RAiSD-AI executable file
    echo "Checking for $RAiSD_AI_VERSION executable..."
    if [ ! -f "$RAiSD_AI_VERSION" ]; then
        echo "$RAiSD_AI_VERSION executable not found."
        echo "Compiling $RAiSD_AI_VERSION..."
	source $(conda info --base)/bin/activate raisd-ai
#	$PACKAGE_MANAGER activate raisd-ai
        ./compile-$RAiSD_AI_VERSION.sh
    else
        echo "$RAiSD_AI_VERSION executable found."
        # ask for confirmation to recompile
        read -p "Do you want to recompile $RAiSD_AI_VERSION? (y/n) " answer
        if [[ "$answer" == "y" ]]; then
            echo "Recompiling $RAiSD_AI_VERSION..."
	    source $(conda info --base)/bin/activate base
            ./compile-$RAiSD_AI_VERSION.sh
        else
            echo "Skipping compilation."
        fi
    fi

}

while getopts "hap:r" opt; do
    case $opt in
        h)
        echo "Usage: $0 [-h] [-a] [-p] [-r]"
        echo "  -h  Show this help message and exit"
        echo "  -a  Install the dependencies for both RAiSD-AI and its GUI"
        echo "  -p  Specify package manager (e.g., conda, micromamba)"
        echo "  -r  Compile RAiSD-AI instead of RAiSD-AI-ZLIB (not recommended)"
        exit 0
        ;;
        a)
        INSTALL_ALL=true
        ;;
        p)
        # Package manager set
        PACKAGE_MANAGER=$OPTARG
        ;;
        r)
        # Compile RAiSD-AI instead of RAiSD-AI-ZLIB
        RAiSD_AI_VERSION="RAiSD-AI"
        ;;
        *)
        echo "Invalid option: -$OPTARG" >&2
        exit 1
        ;;
    esac
    done

if $INSTALL_ALL; then
    echo "Installing $RAiSD_AI_VERSION and its environment..."
    create_environment "environment-raisd-ai.yml"
    install_raisd_ai
fi

echo "Setting up RAiSD-AI-GUI environment..."
create_environment "environment-raisd-ai-gui.yml"
```

**Comment by Steef-Broeder** [comment]
Updated packages:
(added, gcc, make, libpng)

# packages in environment at /home/raisd/miniconda3/envs/raisd-ai:
#
# Name                       Version          Build                  Channel
_libgcc_mutex                0.1              conda_forge            conda-forge
_openmp_mutex                4.5              2_gnu                  conda-forge
_tflow_select                2.3.0            mkl
abseil-cpp                   20211102.0       hd4dd3e8_0
absl-py                      2.1.0            py38h06a4308_0
aiohttp                      3.9.5            py38h5eee18b_0
aiosignal                    1.2.0            pyhd3eb1b0_0
appdirs                      1.4.4            pyh9f0ad1d_0           conda-forge
asciitree                    0.3.3            py_2                   conda-forge
astunparse                   1.6.3            py_0
async-timeout                4.0.3            py38h06a4308_0
attrs                        24.2.0           pyh71513ae_0           conda-forge
bcftools                     1.17             h3cc50cf_1             bioconda
binutils_impl_linux-64       2.40             ha1999f0_7             conda-forge
blas                         1.0              openblas               conda-forge
blinker                      1.6.2            py38h06a4308_0
brotli                       1.0.9            he6710b0_2
brotli-python                1.1.0            py38h17151c0_1         conda-forge
bzip2                        1.0.8            h4bc722e_7             conda-forge
c-ares                       1.33.1           heb4867d_0             conda-forge
ca-certificates              2024.9.24        h06a4308_0
cachetools                   5.3.3            py38h06a4308_0
certifi                      2024.8.30        py38h06a4308_0
cffi                         1.17.0           py38heb5c249_0         conda-forge
charset-normalizer           3.3.2            pyhd8ed1ab_0           conda-forge
click                        8.1.7            py38h06a4308_0
contourpy                    1.0.5            py38hdb19cb5_0
cryptography                 41.0.3           py38h130f0dd_0
cudatoolkit                  11.8.0           h6a678d5_0
cycler                       0.11.0           pyhd3eb1b0_0
cyrus-sasl                   2.1.28           h9c0eb46_1
dbus                         1.13.18          hb2f20db_0
demes                        0.2.3            pyhd8ed1ab_0           conda-forge
expat                        2.6.3            h6a678d5_0
fasteners                    0.17.3           pyhd8ed1ab_0           conda-forge
filelock                     3.13.1           py38h06a4308_0
flatbuffers                  24.3.25          h6a678d5_0
fontconfig                   2.14.1           hef1e5e3_0
fonttools                    4.51.0           py38h5eee18b_0
freetype                     2.12.1           h4a9f257_0
frozenlist                   1.4.0            py38h5eee18b_0
gast                         0.5.3            pyhd3eb1b0_0
gcc                          14.1.0           h6f9ffa1_1             conda-forge
gcc_impl_linux-64            14.1.0           h3c94d91_1             conda-forge
giflib                       5.2.1            h5eee18b_3
glib                         2.78.4           h6a678d5_0
glib-tools                   2.78.4           h6a678d5_0
gmp                          6.2.1            h295c915_3
gmpy2                        2.1.2            py38heeb90bb_0
google-auth                  2.29.0           py38h06a4308_0
google-auth-oauthlib         0.4.4            pyhd3eb1b0_0
google-pasta                 0.2.0            pyhd3eb1b0_0
grpc-cpp                     1.46.1           h33aed49_1
grpcio                       1.42.0           py38hce63b2e_0
gsl                          2.7              he838d99_0             conda-forge
gst-plugins-base             1.14.1           h6a678d5_1
gstreamer                    1.14.1           h5eee18b_1
h2                           4.1.0            pyhd8ed1ab_0           conda-forge
h5py                         3.11.0           py38hbe37b52_0
hdf5                         1.12.1           h70be1eb_2
hpack                        4.0.0            pyh9f0ad1d_0           conda-forge
htslib                       1.17             h6bc39ce_1             bioconda
humanize                     4.10.0           pyhd8ed1ab_0           conda-forge
hyperframe                   6.0.1            pyhd8ed1ab_0           conda-forge
icu                          58.2             he6710b0_3
idna                         3.8              pyhd8ed1ab_0           conda-forge
importlib-metadata           8.4.0            pyha770c72_0           conda-forge
importlib_metadata           8.4.0            hd8ed1ab_0             conda-forge
importlib_resources          6.4.4            pyhd8ed1ab_0           conda-forge
jinja2                       3.1.4            py38h06a4308_0
jpeg                         9e               h5eee18b_3
jsonschema                   4.23.0           pyhd8ed1ab_0           conda-forge
jsonschema-specifications    2023.12.1        pyhd8ed1ab_0           conda-forge
kastore                      0.3.2            py38he82f83a_3         conda-forge
keras                        2.8.0            py38h06a4308_0
keras-preprocessing          1.1.2            pyhd3eb1b0_0
kernel-headers_linux-64      6.12.0           he073ed8_6             conda-forge
keyutils                     1.6.1            h166bdaf_0             conda-forge
kiwisolver                   1.4.4            py38h6a678d5_0
krb5                         1.20.1           hf9c8cef_0             conda-forge
ld_impl_linux-64             2.40             hf3520f5_7             conda-forge
libblas                      3.9.0            20_linux64_openblas    conda-forge
libcblas                     3.9.0            20_linux64_openblas    conda-forge
libclang                     14.0.6           default_hc6dbbc7_1
libclang13                   14.0.6           default_he11475f_1
libcups                      2.4.2            ha637b67_0
libcurl                      7.88.1           h91b91d3_2
libdeflate                   1.21             h4bc722e_0             conda-forge
libedit                      3.1.20191231     he28a2e2_2             conda-forge
libev                        4.33             hd590300_2             conda-forge
libffi                       3.4.2            h7f98852_5             conda-forge
libgcc                       14.1.0           h77fa898_1             conda-forge
libgcc-devel_linux-64        14.1.0           h5d3d1c9_101           conda-forge
libgcc-ng                    14.1.0           h69a702a_1             conda-forge
libgfortran                  14.1.0           h69a702a_1             conda-forge
libgfortran-ng               14.1.0           h69a702a_1             conda-forge
libgfortran5                 14.1.0           hc5f4f2c_1             conda-forge
libglib                      2.78.4           hdc74915_0
libgomp                      14.1.0           h77fa898_1             conda-forge
libiconv                     1.16             h5eee18b_3
liblapack                    3.9.0            20_linux64_openblas    conda-forge
libllvm14                    14.0.6           hecde1de_4
libnghttp2                   1.52.0           ha637b67_1
libnsl                       2.0.1            hd590300_0             conda-forge
libopenblas                  0.3.25           pthreads_h413a1c8_0    conda-forge
libpng                       1.6.39           h5eee18b_0
libpq                        12.15            h37d81fd_1
libprotobuf                  3.20.3           he621ea3_0
libsanitizer                 14.1.0           hcba0ae0_1             conda-forge
libsqlite                    3.46.0           hde9e2c9_0             conda-forge
libssh2                      1.10.0           h37d81fd_2
libstdcxx                    14.1.0           hc0a3c3a_1             conda-forge
libstdcxx-ng                 14.1.0           h4852527_1             conda-forge
libtiff                      4.2.0            hecacb30_2
libuuid                      2.38.1           h0b41bf4_0             conda-forge
libwebp-base                 1.3.2            h5eee18b_0
libxcb                       1.15             h7f8727e_0
libxcrypt                    4.4.36           hd590300_1             conda-forge
libxkbcommon                 1.0.1            hfa300c1_0
libxml2                      2.9.14           h74e7548_0
libzlib                      1.2.13           h4ab18f5_6             conda-forge
make                         4.2.1            h1bed415_1
markdown                     3.4.1            py38h06a4308_0
markupsafe                   2.1.3            py38h5eee18b_0
matplotlib                   3.7.1            py38h06a4308_1
matplotlib-base              3.7.1            py38h417a72b_1
mpc                          1.1.0            h10f8cd9_1
mpfr                         4.0.2            hb69a4c5_1
mpmath                       1.3.0            py38h06a4308_0
msgpack-python               1.0.8            py38hea7755e_0         conda-forge
msprime                      1.2.0            py38h313c355_3         conda-forge
multidict                    6.0.4            py38h5eee18b_0
mysql                        5.7.24           hbb652a2_0
ncurses                      6.5              he02047a_1             conda-forge
networkx                     3.1              py38h06a4308_0
ninja                        1.10.2           h06a4308_5
ninja-base                   1.10.2           hd09550d_5
numcodecs                    0.12.1           py38h854fd01_1         conda-forge
numpy                        1.24.4           py38h59b608b_0         conda-forge
oauthlib                     3.2.2            py38h06a4308_0
olefile                      0.46             pyhd3eb1b0_0
openblas                     0.3.25           pthreads_h7a3da1a_0    conda-forge
openssl                      1.1.1w           h7f8727e_0
opt_einsum                   3.3.0            pyhd3eb1b0_1
packaging                    24.1             pyhd8ed1ab_0           conda-forge
pandas                       2.0.3            py38h01efb38_1         conda-forge
pcre2                        10.42            hebb0a14_1
perl                         5.32.1           7_hd590300_perl5       conda-forge
pillow                       7.0.0            py38hb39fc2d_0
pip                          24.2             pyhd8ed1ab_0           conda-forge
pkgutil-resolve-name         1.3.10           pyhd8ed1ab_1           conda-forge
platformdirs                 4.2.2            pyhd8ed1ab_0           conda-forge
ply                          3.11             py38_0
pooch                        1.8.2            pyhd8ed1ab_0           conda-forge
protobuf                     3.20.3           py38h6a678d5_0
pyasn1                       0.4.8            pyhd3eb1b0_0
pyasn1-modules               0.2.8            py_0
pycparser                    2.22             pyhd8ed1ab_0           conda-forge
pyjwt                        2.8.0            py38h06a4308_0
pyopenssl                    23.2.0           py38h06a4308_0
pyparsing                    3.1.4            pyhd8ed1ab_0           conda-forge
pyqt                         5.15.10          py38h6a678d5_0
pyqt5-sip                    12.13.0          py38h5eee18b_0
pyslim                       1.0.4            pyhd8ed1ab_0           conda-forge
pysocks                      1.7.1            pyha2e5f31_6           conda-forge
python                       3.8.15           h7a1cb2a_2
python-dateutil              2.9.0            pyhd8ed1ab_0           conda-forge
python-flatbuffers           24.3.25          py38h06a4308_0
python-tzdata                2024.1           pyhd8ed1ab_0           conda-forge
python_abi                   3.8              2_cp38                 conda-forge
pytorch                      2.0.1            cpu_py38hab5cca8_0
pytz                         2024.1           pyhd8ed1ab_0           conda-forge
pyyaml                       6.0.1            py38h5eee18b_0
qt-main                      5.15.2           h5b8104b_9
re2                          2022.04.01       h295c915_0
readline                     8.2              h8228510_1             conda-forge
referencing                  0.35.1           pyhd8ed1ab_0           conda-forge
requests                     2.32.3           pyhd8ed1ab_0           conda-forge
requests-oauthlib            2.0.0            py38h06a4308_0
rpds-py                      0.20.0           py38h4005ec7_0         conda-forge
rsa                          4.7.2            pyhd3eb1b0_1
ruamel.yaml                  0.18.6           py38h01eb140_0         conda-forge
ruamel.yaml.clib             0.2.8            py38h01eb140_0         conda-forge
scipy                        1.10.1           py38h59b608b_3         conda-forge
setuptools                   72.2.0           pyhd8ed1ab_0           conda-forge
sip                          6.7.12           py38h6a678d5_0
six                          1.16.0           pyh6c4a22f_0           conda-forge
slim                         4.1              h59595ed_0             conda-forge
snappy                       1.2.1            h6a678d5_0
sqlite                       3.45.3           h5eee18b_0
stdpopsim                    0.2.0            pyhd8ed1ab_0           conda-forge
svgwrite                     1.4.3            pyhd8ed1ab_0           conda-forge
sympy                        1.12             py38h06a4308_0
sysroot_linux-64             2.39             hc4b9eeb_6             conda-forge
tensorboard                  2.8.0            py38h06a4308_0
tensorboard-data-server      0.6.1            py38h52d8a92_0
tensorboard-plugin-wit       1.8.1            py38h06a4308_0
tensorflow                   2.8.2            mkl_py38hb41d75a_0
tensorflow-base              2.8.2            mkl_py38hf890080_0
tensorflow-estimator         2.8.0            py38hb070fc8_0
termcolor                    2.1.0            py38h06a4308_0
tk                           8.6.13           noxft_h4845f30_101     conda-forge
tomli                        2.0.1            py38h06a4308_0
torchvision                  0.15.2           cuda118py38h196c800_0
tornado                      6.4.1            py38h5eee18b_0
tskit                        0.5.6            py38he82f83a_2         conda-forge
typing-extensions            4.11.0           py38h06a4308_0
typing_extensions            4.11.0           py38h06a4308_0
tzdata                       2025c            hc9c84f9_1             conda-forge
unicodedata2                 15.1.0           py38h5eee18b_0
urllib3                      2.2.2            pyhd8ed1ab_1           conda-forge
werkzeug                     3.0.3            py38h06a4308_0
wheel                        0.44.0           pyhd8ed1ab_0           conda-forge
wrapt                        1.14.1           py38h5eee18b_0
xz                           5.2.6            h166bdaf_0             conda-forge
yaml                         0.2.5            h7b6447c_0
yarl                         1.9.3            py38h5eee18b_0
zarr                         2.17.1           pyhd8ed1ab_0           conda-forge
zipp                         3.20.1           pyhd8ed1ab_0           conda-forge
zlib                         1.2.13           h4ab18f5_6             conda-forge
zstandard                    0.23.0           py38h62bed22_0         conda-forge
zstd                         1.5.6            ha6fb4c9_0             conda-forge

File: README.md [hunk 1]
```diff
@@ -75,14 +75,21 @@ The executable is placed in the path bin/release. A link to the executable is pl
...
+### Installing the GUI
 
-The following commands can be used to install all necessary tools (including micromamba for package management), create the environments, and build RAiSD-AI.
+RAiSD-AI includes an optional graphical user interface (GUI).
 
-```bash
-./setup-step-1.sh
-./setup-step-2.sh
-```
+Use the provided `setup-gui.sh` script to install the  dependencies of the gui. To see all options use the help menu (`./setup-gui.sh -h`):
```
Comment (VladNegara) [inline README.md]:
```suggestion
Run the provided `setup-gui.sh` script to install the dependencies of the GUI. To see all options, use the help option: `./setup-gui.sh -h`.
```

Adding this to the suggestion breaks the GitHub comment, but maybe add `./setup-gui.sh` in a code block between the two sentences to make it as clear as possible to the user what they need to type in the terminal. The `./setup-gui.sh -h` could then also be brought down to its own line in a code block for consistency.
